core/environment/environment.go

```diff
@@ -602,6 +602,7 @@ func newEnvironment(userVars map[string]string, newId uid.ID) (env *Environment,
 							Debug("O2 End Completion time already set before after_GO_ERROR")
 					}
 					env.invalidateAutoStopTransition()
+					env.currentRunNumber = 0
```

Comment:
20 lines lower there is this:
```
				if e.Event == "STOP_ACTIVITY" {
					// If the event is STOP_ACTIVITY, we remove the active run number after all hooks are done.
					env.workflow.GetVars().Set("last_run_number", strconv.Itoa(int(env.currentRunNumber)))
					env.currentRunNumber = 0
					env.workflow.GetVars().Del("run_number")
					env.workflow.GetVars().Del("runNumber")
				}
	```
Shouldn't we just modify it so that it also deletes run numbers in case of GO_ERROR? We have to keep in mind though that GO_ERROR may not necessarily happen during RUNNING, but also in other states.

Reply:
I am suggesting this, because the GO_ERROR hooks with positive weights should still have access to run number.